### PR TITLE
Add CountryFlag component and flags assets

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

First version of the app! :-)

This adds the `CountryFlag` component and the flags assets for all countries.

[Related clubhouse story](https://app.clubhouse.io/vtex/story/38073/create-app-to-host-all-country-flags)

**Note to reviewers**: The only "relevant" files are the `react/CountryFlag.tsx` and `react/flagMap.tsx`, you can ignore the other ones as they only are the components with the flags SVGs.

#### How should this be manually tested?

[Workspace](https://lucas--checkoutio.myvtex.com/checkout/#/profile).

The app is currently being used by both the phone number input (profile) and the country selector (shipping).

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

image

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/flagMap.tsx [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+const YEM = React.lazy(() => import('./icons/YEM'))
+const ZAF = React.lazy(() => import('./icons/ZAF'))
+const ZMB = React.lazy(() => import('./icons/ZMB'))
+const ZWE = React.lazy(() => import('./icons/ZWE'))
+
+/**
+ * Mapping of country's ISO-3 code to the flag component
+ */
+const flagMap: { [key: string]: React.ComponentType } = {
+  AFG,
```
Comment (marcoskwkm) [inline react/flagMap.tsx]: Does it work if you do something like this?
```suggestion
  AFG: React.lazy(() => import('./icons/AFG'))
```

Having all the country codes twice in the code bothers me a bit.

Comment (lucasecdb) [reply react/flagMap.tsx]: sure! indeed it's better than have it duplicated in the same file

Comment (lucasecdb) [reply react/flagMap.tsx]: done in 1202552

**Review by marcoskwkm — APPROVED** [review]

File: react/flagMap.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+export function loadFlag(countryCode?: CountryCode): React.ReactElement | null {
+  if (!countryCode || !flagMap[countryCode]) {
+    return null
+  }
+
+  if (typeof window === 'undefined') {
+    return <div style={{ height: 24, width: 24 }} />
+  }
+
+  const Flag = flagMap[countryCode]
```
Comment (Klynger) [inline react/flagMap.tsx]:
```suggestion
  const Flag = flagMap[countryCode]
  if (typeof window === 'undefined' || !Flag) {
    return <div style={{ height: 24, width: 24 }} />
  }
```

Comment (lucasecdb) [reply react/flagMap.tsx]: actually we need to return `null` here if there is no flag component, thanks for the reminder!

**Review by Klynger — APPROVED** [review]

**Comment by lucasecdb** [comment]
I'll hold the release until we are back to DEFCON 5

**Comment by lucasecdb** [comment]
damn, I didn't configure the bot for this repo 🤦‍♂️

**Pull request merged**